api/envoy/config/cluster/v3/outlier_detection.proto

```diff
@@ -173,4 +173,8 @@ message OutlierDetection {
   // Set of host's passive monitors.
   // [#not-implemented-hide:]
   repeated core.v3.TypedExtensionConfig monitors = 24;
+
+  // If may_always_eject_one is enabled, :ref:`base_ejection_time<envoy_v3_api_field_config.cluster.v3.OutlierDetection.max_ejection_percent>` is not considered for first ejection.
```

Comment:
nit: s/If may_always_eject_one is enabled/If enabled

Reply:
Also, can you explain what you mean by "is not considered for first ejection"?  Does that mean the first host in the host list for the upstream cluster will be removed regardless, if this flag is true?

Reply:
yeah, first host will be ejected regardless of mep

Reply:
> Does that mean the first host in the host list for the upstream cluster will be removed regardless, if this flag is true?

Actually no, not the first one from the list in upstream cluster. That host may be totally healthy. It means that if a particular host failed outlier detection check, it will be ejected, even when the ejection would break other limits, like max_percentage of host which may be ejected.

Reply:
sorry, I misread that. Yes, first host that fails outlier detection check will be ejected (doesn't have to be first in the list)